tools/py/add_mod_poly_c.pyx

```diff
@@ -0,0 +1,128 @@
+from libc.stdio cimport printf
```

Comment:
can't we factorize code with the pure python code ?
Or at least separate in another directory

Reply:
in Cython you select the functions that can be called from python (def ou cpdef functions)
here test_full_mod is the "allowed" python function (even if it calls cdef functions that are pure C and have no call overhead)
test_full_mod function is imported in the pure py in /https://github.com/keep-starknet-strange/garaga/blob/b02bb7dacf677939a08e0ff99cbdf39180a56dd8/tests/python_tests/add_mod_poly_extensive_test.py
overall .pyx files, after being compiled, acts (if you define them so with cpdef or def) as C-compiled libraries that can directly be imported

Reply:
However, add_mod_poly.py will be deleted and replaced by this .pyx file in the future for cleaner tools/py folder